catalog/models.py

```diff
@@ -0,0 +1,163 @@
+from django.db import models
+from django.urls import reverse # Used to generate URLs by reversing the URL patterns
+import uuid # Required for unique book copies
+from django.contrib.auth.models import User
+from datetime import date
+
+from django.core.validators import MaxValueValidator, MinValueValidator
+from django.db.models import Avg
+
+class Genre(models.Model):
+    """Model representing a book genre."""
+    name = models.CharField(max_length=200, help_text='Enter a book genre (e.g. Science Fiction)')
+
+    def __str__(self):
+        """String for representing the Model object."""
+        return self.name
+
+class Language(models.Model):
+    """Model representing a Language (e.g. English, French, Japanese, etc.)"""
+    name = models.CharField(max_length=200,
+                            help_text="Enter the book's natural language (e.g. English, French, Japanese etc.)")
+
+    def __str__(self):
+        """String for representing the Model object (in Admin site etc.)"""
+        return self.name
+
+class Review(models.Model):
+    user = models.ForeignKey(User, on_delete=models.SET_NULL, null=True, blank=False)
+    book = models.ForeignKey('Book', on_delete=models.CASCADE)
+    comment = models.TextField(max_length=500, help_text="Enter comment about the book", null=True, blank=True)
+    point = models.IntegerField(
+        default=5,
+        validators=[
+            MaxValueValidator(5),
+            MinValueValidator(1)
+        ],
+        null=False,
+        blank=False)
+    created_at = models.DateTimeField(auto_now_add=True)
+
+class Book(models.Model):
+    """Model representing a book (but not a specific copy of a book)."""
+    title = models.CharField(max_length=200)
+    author = models.ForeignKey('Author', on_delete=models.SET_NULL, null=True)
+    summary = models.TextField(max_length=1000, help_text='Enter a brief description of the book')
+    isbn = models.CharField('ISBN', max_length=13, unique=True,
+                            help_text='13 Character <a href="https://www.isbn-international.org/content/what-isbn">ISBN number</a>')
+    genre = models.ManyToManyField(Genre, help_text='Select a genre for this book')
+    language = models.ForeignKey('Language', on_delete=models.SET_NULL, null=True)
+
+    def __str__(self):
+        """String for representing the Model object."""
+        return self.title
+
+    def get_absolute_url(self):
+        """Returns the URL to access a detail record for this book."""
+        return reverse('book-detail', args=[str(self.id)])
+
+    def display_genre(self):
+        """Create a string for the Genre. This is required to display genre in Admin."""
+        return ', '.join(genre.name for genre in self.genre.all()[:3])
+
+    display_genre.short_description = 'Genre'
+
+    def get_average_rating(self):
+        return (Review.objects.all()
+            .filter(book=self)
+            .aggregate(avg_rating=Avg('point'))
```

Comment:
Nên lưu lại giá trị trung bình vào DB và cập nhật nó mỗi khi có thay đổi rating, thay vì đi tính toán mỗi lần get dữ liệu e nhé